src/guardian/guardian.service.ts

```diff
@@ -202,16 +258,15 @@ export class GuardianService implements OnModuleInit {
   public async checkKeysIntersections(blockData: BlockData): Promise<void> {
     const { blockHash } = blockData;
 
-    const nextKeysIntersections = this.getNextKeysIntersections(blockData);
-    const cachedKeysIntersections = this.getCachedKeysIntersections(blockData);
-    const intersections = nextKeysIntersections.concat(cachedKeysIntersections);
+    const keysIntersections = this.getKeysIntersections(blockData);
+
     const filteredIntersections = await this.excludeEligibleIntersections(
-      intersections,
+      keysIntersections,
       blockData,
     );
     const isFilteredIntersectionsFound = filteredIntersections.length > 0;
 
-    this.collectIntersectionsMetrics(intersections, filteredIntersections);
+    this.collectIntersectionsMetrics(keysIntersections, filteredIntersections);
```

Comment:
this is called in the module loop, so the metrics will be overwritten by the last processed module